servo_gtk/web_view.rs

```diff
@@ -477,40 +240,42 @@ impl WebView {
         };
 
         match event_type {
-            // FIXME: this is just a hack to get me going. Ideally we would
-            // use a DMA-Buf so we avoid movign the pixels from the GPU to
-            // system memory and back to the GPU
             servo_event::Event::FrameReady(frame_ready) => {
                 let rgba_image = RgbaImage::from_raw(
                     frame_ready.width,
                     frame_ready.height,
                     frame_ready.rgba_data,
```

Comment:
uh oh, why does `servo_event` already return CPU pixels? 🤔 maybe it's not the right level of integration for this.

You'd only need dmabuf if you'd extracted the engine into an external process; if you render in-process with GL, you can use GdkGLTextureBuilder to get a paintable that directly references your texture. See for example [this](https://gitlab.gnome.org/jpu/casilda/-/commit/33b207aea28bad1130157618f5df1165dc008e2f).